### PR TITLE
xds: ensure the active streams counters are 64 bit aligned on 32 bit systems

### PR DESCRIPTION
**Pull request by rboyer**

### CODE REVIEW

**Review by mkeeler — APPROVED** [review]

File: agent/xds/server.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ type Server struct {
 
 	DisableV2Protocol bool
 
-	activeStreams activeStreamCounters
+	activeStreams *activeStreamCounters
 }
 
 // activeStreamCounters simply encapsulates two counters accessed atomically to
```
Comment (kisunji) [inline agent/xds/server.go:158]: Is this comment still accurate? Maybe we could add a small explanation why `activeStreams` is a pointer

**Review by kisunji — APPROVED** [review]

File: agent/xds/server.go (hunk 1 above)
Comment (rboyer) [reply agent/xds/server.go:158]: Yeah I guess I should adjust that comment slightly.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/450123.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 5fe613dd05b05ed0d9e5d0bc3cf1b96368d00e73 onto `release/1.11.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 5fe613dd05b05ed0d9e5d0bc3cf1b96368d00e73 onto `release/1.10.x` succeeded!

**Pull request merged**